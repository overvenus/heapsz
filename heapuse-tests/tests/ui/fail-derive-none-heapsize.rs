use heapuse_derive::Heap;

struct NoneHeapSize {
    a: ()
}

#[derive(Heap)]
pub struct BadStruct {
    #[heap(add)]
    pub a: NoneHeapSize,
}

fn main() {}
