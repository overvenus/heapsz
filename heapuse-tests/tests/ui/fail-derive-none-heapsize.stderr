error[E0277]: the trait bound `NoneHeapSize: HeapSize` is not satisfied
  --> tests/ui/fail-derive-none-heapsize.rs:9:5
   |
9  | /     #[heap(add)]
10 | |     pub a: NoneHeapSize,
   | |_______________________^ the trait `HeapSize` is not implemented for `NoneHeapSize`
   |
   = help: the following other types implement trait `HeapSize`:
             bool
             isize
             i8
             i16
             i32
             i64
             i128
             usize
           and $N others
